tests/epyccel/recognised_functions/test_numpy_types.py

```diff
@@ -9,9 +9,121 @@
 from test_numpy_funcs import max_float, min_float, max_float32, min_float32,max_float64, min_float64
 from test_numpy_funcs import matching_types
 
-from pyccel.decorators import types
+from pyccel.decorators import types, template
 from pyccel.epyccel import epyccel
 
+def test_mult_numpy_python_type(language):
+
+    def mult_on_array_int8():
+        from numpy import ones, int8
+        a = ones(5, dtype=int8)
+        b = a * 2
+        return b[0]
+
+    def mult_on_array_int16():
+        from numpy import ones, int16
+        a = ones(5, dtype=int16)
+        b = a * 2
+        return b[0]
+
+    def mult_on_array_int32():
+        from numpy import ones, int32
+        a = ones(5, dtype=int32)
+        b = a * 2
+        return b[0]
+
+    def mult_on_array_int64():
+        from numpy import ones, int64
+        a = ones(5, dtype=int64)
+        b = a * 2
+        return b[0]
+
+    def mult_on_array_float32():
+        from numpy import ones, float32
+        a = ones(5, dtype=float32)
+        b = a * 2
+        return b[0]
+
+    def mult_on_array_float64():
+        from numpy import ones, float64
+        a = ones(5, dtype=float64)
+        b = a * 2
+        return b[0]
+
+    epyccel_func = epyccel(mult_on_array_int8, language=language)
+    python_result = mult_on_array_int8()
+    pyccel_result = epyccel_func()
+    assert matching_types(pyccel_result, python_result)
+
+    epyccel_func = epyccel(mult_on_array_int16, language=language)
+    python_result = mult_on_array_int16()
+    pyccel_result = epyccel_func()
+    assert matching_types(pyccel_result, python_result)
+
+    epyccel_func = epyccel(mult_on_array_int32, language=language)
+    python_result = mult_on_array_int32()
+    pyccel_result = epyccel_func()
+    assert matching_types(pyccel_result, python_result)
+
+    epyccel_func = epyccel(mult_on_array_int64, language=language)
+    python_result = mult_on_array_int64()
+    pyccel_result = epyccel_func()
+    assert matching_types(pyccel_result, python_result)
+
+    epyccel_func = epyccel(mult_on_array_float32, language=language)
+    python_result = mult_on_array_float32()
+    pyccel_result = epyccel_func()
+    assert matching_types(pyccel_result, python_result)
+
+    epyccel_func = epyccel(mult_on_array_float64, language=language)
+    python_result = mult_on_array_float64()
+    pyccel_result = epyccel_func()
+    assert matching_types(pyccel_result, python_result)
```

Comment:
It seems a shame not to also test the result in this test function

Reply:
Surely, i will add that.

Reply:
Done in [931707b](https://github.com/pyccel/pyccel/pull/1187/commits/931707bcf90417a380050074300fa42212e35785).